app/controllers/sks_users_controller.ts

```diff
@@ -0,0 +1,121 @@
+import { HttpContext } from '@adonisjs/core/http'
+import SksUser from '#models/sks_user'
+import { DateTime } from 'luxon'
+
+//this value determines the period over which the trend will be counted (multiply the value by 5 minutes)
+const trendDelta = 3
+
+enum Trend {
+  INCREASING = 'INCREASING',
+  DECREASING = 'DECREASING',
+  STABLE = 'STABLE',
+}
+
+export default class SksUsersController {
+  /**
+   * Display the latest SKS users record based on time
+   */
+  async latest({ response }: HttpContext) {
+    try {
+      const currentTime = DateTime.now().setZone('Europe/Warsaw').toSQL()
+      if (currentTime === null) {
+        return response.status(500).json({ message: 'Failed to convert time to SQL format' })
+      }
+
+      let latestData = await SksUser.query()
+        .where('externalTimestamp', '<', currentTime)
+        .orderBy('externalTimestamp', 'desc')
+        .first()
+      let isResultRecent = true
```

Comment:
too much `let` variables in here, this often means you can do a refactor for better code (not always but it can be some indicator).  I beliece smth like this would be nicer:
```ts
const latestData = await SksUser.query()
        .where('externalTimestamp', '<', currentTime)
        .orderBy('externalTimestamp', 'desc')
        .first()
const isResultRecent = latestData !== null && latestData.activeUsers > 0
// If the first record has activeUsers set to 0, get the second record instead
const entryToReturn = isResultRecent ? latestData :  await SksUser.query()
          .where('externalTimestamp', '<', currentTime)
          .orderBy('externalTimestamp', 'desc')
          .offset(1)
          .first()
```